src/modules/ModuleRunner.ts

```diff
@@ -31,91 +31,80 @@ import { ModuleFactory } from "./ModuleFactory";
 
 import "./ModuleComponents";
 
-class ExtensionImplementationMap {
-    public hasDefaultCryptoSetupExtension: boolean = true;
-    public hasDefaultExperimentalExtension: boolean = true;
-}
-
 /**
- * Handles and manages any extensions provided by modules
+ * Handles and manages extensions provided by modules.
  */
 class ExtensionsManager {
-    private _cryptoSetup: ProvideCryptoSetupExtensions;
-    private _experimental: ProvideExperimentalExtensions;
-    private _implementionMap: ExtensionImplementationMap = new ExtensionImplementationMap();
+    // Private backing fields for extensions
+    private cryptoSetupExtension: ProvideCryptoSetupExtensions;
+    private experimentalExtension: ProvideExperimentalExtensions;
+
+    // Map to keep track of which extensions has been provided by modules.
+    private implementionMap = {
+        hasDefaultCryptoSetupExtension: true,
+        hasDefaultExperimentalExtension: true,
+    };
 
```

Comment:
I think we should flatten this structure further:

```suggestion
    /** `true` if `cryptoSetupExtension` is the default implementation; `false` if it is implemented by a module. */
    private hasDefaultCryptoSetupExtension = true;

    /** `true` if `experimentalExtension` is the default implementation; `false` if it is implemented by a module. */    
    private hasDefaultExperimentalExtension: true,
    };

```

Reply:
OK. Done